### PR TITLE
SONARSCGIT-13 Detect shallow clone and fail fast

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/JGitBlameCommand.java [hunk 1]
```diff
@@ -53,6 +54,9 @@ public void blame(BlameInput input, BlameOutput output) {
     File basedir = input.fileSystem().baseDir();
     try (Repository repo = buildRepository(basedir); Git git = Git.wrap(repo)) {
       File gitBaseDir = repo.getWorkTree();
+      if (gitBaseDir.toPath().resolve(".git/shallow").toFile().isFile()) {
```
Comment (dbmeneses) [inline sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/JGitBlameCommand.java]: You could also use Files.isRegularFile

File: sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/JGitBlameCommand.java [hunk 2]
```diff
@@ -53,6 +54,9 @@ public void blame(BlameInput input, BlameOutput output) {
...
+        throw MessageException.of("Shallow clone detected, blame information may be incorrect. You can convert to non-shallow with: git fetch --unshallow");
```
Comment (dbmeneses) [inline sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/JGitBlameCommand.java]: What about  also informing that scm blame can be disabled?

File: sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/JGitBlameCommand.java [hunk 3]
```diff
@@ -53,6 +55,9 @@ public void blame(BlameInput input, BlameOutput output) {
     File basedir = input.fileSystem().baseDir();
     try (Repository repo = buildRepository(basedir); Git git = Git.wrap(repo)) {
       File gitBaseDir = repo.getWorkTree();
+      if (Files.isRegularFile(gitBaseDir.toPath().resolve(".git/shallow"))) {
```
Comment (SonarTech) [inline sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/JGitBlameCommand.java]: MAJOR Replace this with a call to the "toFile().isFile()" method [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**